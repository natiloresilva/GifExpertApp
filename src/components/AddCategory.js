import React, { useState } from 'react'
import propTypes from 'prop-types'

export const AddCategory = ({setCategories, categories}) => {

    const [inputValue, setInputValue] = useState('')
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
       
        if(inputValue.trim().length > 2){
            //Si no mandaba las categories por props también podría haber hecho ===> 
            setCategories( cates => [ inputValue, ...cates])
            //setCategories([inputValue, ...categories])
            setInputValue('')
        }
        
       
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
       </form>
    )
}


AddCategory.propTypes = {
    setCategories: propTypes.func.isRequired
}
