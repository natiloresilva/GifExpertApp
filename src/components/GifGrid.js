import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import {GifGridItem} from '../components/GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    //Renombramos Data en la desestructuraci√≥n
    const {data:images, loading} = useFetchGifs(category)

    return (
        <>
        <h3 className='animate__animated animate__fadeIn'>{category}</h3>

        {loading && <p className='animate__animated animate__flash'>Loading ...</p>}
            <div className='card-grid'>
               {
                images.map(img => (
                    <GifGridItem 
                        key={img.id}
                        {...img}
                    />
                ))
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

