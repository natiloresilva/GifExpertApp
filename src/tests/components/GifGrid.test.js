import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {

    test('Debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={'Simpsons'}/>)
        
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/img.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={'Simpsons'}/>)

        //Que haga match con el Snapshot
        expect(wrapper).toMatchSnapshot()

        //Que el <p>Loading...</p> no exista
        expect(wrapper.find('p').exists()).toBe(false)

        //Que haya tantos componentes GifGridItem como gifs en el array
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        
    })
})
