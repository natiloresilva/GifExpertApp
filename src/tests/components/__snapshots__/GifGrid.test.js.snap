// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Simpsons
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Simpsons
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/img.jpg"
    />
  </div>
</Fragment>
`;
